File: tests/output/jpa/src/main/javagen/restaurant/jpa_sequence_metamodel/dtos/restaurant/LigneCommandeItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_sequence_metamodel.dtos.restaurant;

import java.math.BigDecimal;

import jakarta.annotation.Generated;

/**
 * Détail d'une ligne de commande en liste.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public interface LigneCommandeItem {

	/**
	 * Identifiant de la ligne.
	 */
	Integer getId();

	/**
	 * Quantité commandée.
	 */
	Integer getQuantite();

	/**
	 * Prix unitaire au moment de la commande.
	 */
	BigDecimal getPrixUnitaire();

	/**
	 * Prix total de la ligne.
	 */
	BigDecimal getPrixTotal();

	/**
	 * Commande à laquelle appartient la ligne.
	 */
	Integer getCommandeId();

	/**
	 * Plat commandé.
	 */
	Integer getPlatId();

	/**
	 * Hydrate values of instance.
	 * @param id value to set.
	 * @param quantite value to set.
	 * @param prixUnitaire value to set.
	 * @param prixTotal value to set.
	 * @param commandeId value to set.
	 * @param platId value to set.
	 */
	void hydrate(Integer id, Integer quantite, BigDecimal prixUnitaire, BigDecimal prixTotal, Integer commandeId, Integer platId);
}

File: tests/output/focus/src/model/restaurant/ligne-commande-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {e, entity, EntityToType} from "@focus4/entities";
import {DO_ID, DO_PRIX, DO_QUANTITE} from "../../domains";

export type LigneCommandeItem = EntityToType<LigneCommandeItemEntityType>;
export type LigneCommandeItemEntityType = typeof LigneCommandeItemEntity;

export const LigneCommandeItemEntity = entity({
    id: e.field(DO_ID, f => f.optional()
        .label("restaurant.ligneCommande.id")
    ),
    quantite: e.field(DO_QUANTITE, f => f
        .label("restaurant.ligneCommande.quantite")
    ),
    prixUnitaire: e.field(DO_PRIX, f => f
        .label("restaurant.ligneCommande.prixUnitaire")
    ),
    prixTotal: e.field(DO_PRIX, f => f
        .label("restaurant.ligneCommande.prixTotal")
    ),
    commandeId: e.field(DO_ID, f => f
        .label("restaurant.ligneCommande.commandeId")
    ),
    platId: e.field(DO_ID, f => f
        .label("restaurant.ligneCommande.platId")
    )
});
